backend/app/settle_usage.go

```diff
@@ -0,0 +1,191 @@
+	// Define the end of the day (next day at 00:00)
+	endOfDay := time.Date(now.Year(), now.Month(), now.Day()+1, 0, 0, 0, 0, time.Local)
+
+	// Get the last calculation time for this user from the database, or use a default if not available
+	lastCalcTime, err := h.db.GetUserLastCalcTime(userID)
+	if err != nil {
+		return 0, err
+	}
+
+	// If this is the first time or no record exists, use the start of the day as default
+	if lastCalcTime.IsZero() {
+		lastCalcTime = time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, time.Local)
+	}
+
+	contracts, err := h.db.ListAllContractsInPeriod(userID, lastCalcTime, endOfDay)
+	if err != nil {
+		return 0, err
+	}
+
+	if len(contracts) == 0 {
+		return 0, nil
+	}
+
+	var totalDailyUsageInUSDMillicent uint64
+
+	for _, record := range contracts {
+		// Get bill reports from the last calculation time to the end of day
+		billReports, err := internal.ListContractBillReports(h.graphqlClient, record.ContractID, lastCalcTime, endOfDay)
+		if err != nil {
+			return 0, err
+		}
+
+		totalAmountBilledInUSDMillicent, err := h.calculateTotalUsageOfReportsInUSDMillicent(billReports.Reports)
+		if err != nil {
+			return 0, err
+		}
+
+		totalDailyUsageInUSDMillicent += totalAmountBilledInUSDMillicent
+	}
+
+	// Update the last calculation time for this user in the database
+	if err := h.db.UpdateUserLastCalcTime(userID, now); err != nil {
+		logger.GetLogger().Error().Err(err).Msgf("Failed to update last calculation time for user %d", userID)
+	}
+
+	return totalDailyUsageInUSDMillicent, nil
+}
+
+func (h *Handler) calculateTotalUsageOfReportsInUSDMillicent(reports []internal.Report) (uint64, error) {
+	var totalAmountBilledInUSDMillicent uint64
+	for _, report := range reports {
+		amountInTFT, err := removeDiscountFromReport(&report)
+		if err != nil {
+			return 0, err
+		}
+
+		amountInUSDMillicent, err := h.fromTFTtoUSDMillicent(amountInTFT, report)
+		if err != nil {
+			return 0, err
+		}
+
+		totalAmountBilledInUSDMillicent += amountInUSDMillicent
+	}
+
+	return totalAmountBilledInUSDMillicent, nil
+}
+
+func (h *Handler) fromTFTtoUSDMillicent(amount uint64, report internal.Report) (uint64, error) {
+	price, err := h.getBillingRateAt(report)
+	if err != nil {
+		return 0, err
+	}
+
+	usdMillicentBalance := uint64(math.Round((float64(amount) / TFTUnitFactor) * float64(price)))
+	return usdMillicentBalance, nil
+}
+
+func removeDiscountFromReport(report *internal.Report) (uint64, error) {
+	discountPackage := getDiscountPackage(discount(report.DiscountReceived))
+
+	amountBilled, err := strconv.ParseInt(report.AmountBilled, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	amountBilledWithNoDiscount := float64(amountBilled) / float64(1-discountPackage.Discount/100)
+	return uint64(amountBilledWithNoDiscount), nil
+}
+
+func getDiscountPackage(discountInput discount) DiscountPackage {
+	oneDayMargin := 1.0 / 30.0
+
+	discountPackages := map[discount]DiscountPackage{
+		"none": {
+			DurationInMonth: 0,
```

Comment:
The margin is not added here. Also, I think we should use a larger margin for the none package as we will be risking the user workloads in case of any service failure. (maybe 3 days)